lib/adapters/xhr.js

```diff
@@ -63,7 +63,9 @@ module.exports = function xhrAdapter(config) {
 
       // Prepare the response
       var responseHeaders = 'getAllResponseHeaders' in request ? parseHeaders(request.getAllResponseHeaders()) : null;
-      var responseData = !config.responseType || config.responseType === 'text' ? request.responseText : request.response;
+      // IE9 does not provide response if responseType === 'json' (see mzabriskie/axios#265)
+      // In that case tet's try to parse it.
+      var responseData = !config.responseType || config.responseType === 'text' ? request.responseText : ((config.responseType === 'json' && request.response === void 0) ? JSON.parse(request.responseText) : request.response);
```

Comment:
while reworking this nested ternary, please replace `void 0` with `undefined` - we're in strict mode here so undefined is read-only